artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/jaas/AbstractPrincipalLoginModule.java

```diff
@@ -31,12 +31,13 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import java.lang.invoke.MethodHandles;
 
 /**
  * Abstract login module that uses an external authenticated principal
  */
 public abstract class AbstractPrincipalLoginModule implements AuditLoginModule {
-   private final Logger logger = LoggerFactory.getLogger(getClass());
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
Similar to earlier comments, this changes the logger used by the subclasses to all be the same name, making it no longer possible to tell which one the logging related to. Should instead make them pass in their own static logger to retain existing logging behaviour.

Reply:
We should make this standard... it was a mistake to do it like this.


Each class should have it's own final static logger.

Reply:
Basically all this is doing by using getClass() as the key is sharing a logger name (which various places in the codebase do with the code loggers, by using the package name as the key), so that all the logging for a given LoginModule all comes out the same logger name and can be reasoned about as a unit.

If you make this class have a static logger like this youll also need to change all the log messages themselves to better reflect what module they are logging about or you simply wont be able to tell which the logging is about unless there is only one. Passing a logger for such cases is not that unusual, it is done elsewhere in the codebase.

Reply:
It's definitely no use to log.debug on an instance logger... that was a mistake... we had no defined workspace when this was done...


if this was doing log.info() with some information for the user I would agree with you.. .but log.debug() it's information for use on this case....  or someone debugging the code. 

There's no point on doing log.debug on a instance logger on this case...  we should remove them all.


We used to have others that I have historically replaced along the way.. it's just a mistake of stuff we used to do and it should be changed.